core/src/main/scala/kafka/log/remote/RemoteIndexCache.scala

```diff
@@ -152,8 +194,7 @@ class RemoteIndexCache(maxSize: Int = 1024, remoteStorageManager: RemoteStorageM
           val txnIndex = new TransactionIndex(offset, txnIndexFile)
           txnIndex.sanityCheck()
 
-          val entry = new Entry(offsetIndex, timeIndex, txnIndex)
-          entries.put(uuid, entry)
+          internalCache.put(uuid, new Entry(offsetIndex, timeIndex, txnIndex))
```

Comment:
it may not be desirable, but typically you should prefer to use a computation, `cache.get(key, mappingFunction)`, to make the loading atomic and avoid a cache stampede. Otherwise a race could allow duplicate work if not protected elsewhere.

Reply:
Thank you for the suggestion. It is guaranteed that this place in the code (initialization) is always invoked by a single thread. At other place (getIndexEntry()) which requires thread safety, I am using the `get(key, mappingFunction)`.